python/semantic_kernel/connectors/memory/astradb/astra_client.py

```diff
@@ -0,0 +1,159 @@
+import requests
+import json
+from typing import Any, Dict, List, Optional, Union
+
+import aiohttp
+import asyncio
+import nest_asyncio
+nest_asyncio.apply()
+
+class AstraClient:
+    def __init__(
+        self,
+        astra_id: str,
+        astra_region: str,
+        astra_application_token: str,
+        keyspace_name: str,
+        embedding_dim: int,
+        similarity_function: str,
+    ):
+        self.astra_id = astra_id
+        self.astra_application_token = astra_application_token
+        self.astra_region = astra_region
+        self.keyspace_name = keyspace_name
+        self.embedding_dim = embedding_dim
+        self.similarity_function = similarity_function
+
+        self.request_base_url = f"https://{self.astra_id}-{self.astra_region}.apps.astra.datastax.com/api/json/v1/{self.keyspace_name}"
+        self.request_header = {
+            "x-cassandra-token": self.astra_application_token,
+            "Content-Type": "application/json",
+        }
+
+    async def _run_query(self, request_url: str, query: Dict):
+        async with aiohttp.ClientSession() as session:
```

Comment:
I would like to see the option to supply your own session for this, have a look at this PR for how to do that: #4477

Reply:
done